source/verus/src/main.rs

```diff
@@ -75,19 +63,270 @@ fn main() {
+
+    child.wait().map_err(|x| format!("verus failed to run with error: {}", x))?;
+
+    writeln!(file, "{}", err_string)
+        .map_err(|x| format!("failed to write to reports.toml: {}", x))?;
+    writeln!(file, "{}", out_string)
+        .map_err(|x| format!("failed to write to reports.toml: {}", x))?;
+
+    let dep_file_path = reports.dep_path;
+
+    let deps = get_dependencies(&dep_file_path)?;
+
+    let deps_path = deps.iter().map(|x| Path::new(x)).collect::<Vec<_>>();
+
+    // compute common ancester
+    let common = match common_path::common_path_all(deps_path) {
+        Some(common) => common,
+        None => PathBuf::from(""),
+    };
+
+    if deps.clone().into_iter().any(|dep| dep.is_absolute()) {
+        // copy files to repo_path
+        for dep in deps.iter() {
+            let rel_dest = dep.clone().strip_prefix(&common).unwrap().to_path_buf();
+            create_dir_all(proj_path.join(&rel_dest.parent().unwrap())).unwrap();
+            std::fs::copy(dep, proj_path.join(&rel_dest)).map_err(|err| {
+                format!(
+                    "failed to copy file {} to repo_path with error message: {}",
+                    dep.display(),
+                    err
+                )
+            })?;
+        }
+    } else {
+        // copy files to repo_path, none of dep should be absolute path at thsi point
+        for dep in deps.iter() {
+            create_dir_all(proj_path.join(dep.parent().unwrap())).unwrap();
+            std::fs::copy(dep, proj_path.join(dep)).map_err(|err| {
+                format!(
+                    "failed to copy file {} to repo_path with error message: {}",
+                    dep.display(),
+                    err
+                )
+            })?;
         }
     }
+
+    std::fs::remove_file(&dep_file_path).map_err(|err| {
+        format!(
+            "failed to remove dependency file {} in local data cache with error message: {}",
+            dep_file_path.display(),
+            err
+        )
+    })?;
+
+    Command::new("git").current_dir(&proj_path).arg("add").arg(".").output().map_err(|x| {
+        format!("localstorage: failed to stage current project with error message: {}", x)
+    })?;
+
+    std::process::Command::new("git")
+        .current_dir(&proj_path)
+        .arg("commit")
+        .arg("-m")
+        .arg("\"verus\"")
+        .env("GIT_AUTHOR_NAME", "nobody")
+        .env("GIT_AUTHOR_EMAIL", "nobody@nobody.nobody")
+        .env("GIT_COMMITTER_NAME", "nobody")
+        .env("GIT_COMMITTER_EMAIL", "nobody@nobody.nobody")
+        .output()
+        .map_err(|x| format!("localstorage: failed to git commit with error message: {}", x))?;
+
+    Ok(())
+}
+
+#[derive(Debug)]
+pub struct Reports {
+    proj_path: PathBuf,
+    dep_path: PathBuf,
+}
+
+fn repo_path(store: bool) -> Option<Reports> {
+    // check if user has git as executable
+    if Command::new("git").arg("--version").output().is_err() {
+        return None;
+    }
```

Comment:
Assuming we change to the file based opt-in described in another comment, we should not silently stop if the user doesn't have `git` available; instead we should warn (with `eprintln!("NOTE: Opted into Verus analytics, but no `git` available on system. Please make sure git is installed to be able to use analytics")` or similar), before returning `None`.